server/channels/store/sqlstore/job_store.go

```diff
@@ -172,34 +172,84 @@ func (jss SqlJobStore) UpdateStatus(id string, status string) (*model.Job, error
 	return job, nil
 }
 
-func (jss SqlJobStore) UpdateStatusOptimistically(id string, currentStatus string, newStatus string) (bool, error) {
+func (jss SqlJobStore) UpdateStatusOptimistically(id string, currentStatus string, newStatus string) (*model.Job, error) {
+	if jss.DriverName() == model.DatabaseDriverMysql {
+		tx, err := jss.GetMaster().Beginx()
+		if err != nil {
+			return nil, errors.Wrap(err, "begin_transaction")
+		}
+		defer finalizeTransactionX(tx, &err)
+
+		builder := jss.getQueryBuilder().
+			Update("Jobs").
+			Set("LastActivityAt", model.GetMillis()).
```

Comment:
To avoid inconsistencies, I'd probably use a single time value for updating `LastActivityAt` and `StartAt`.

Reply:
Sounds good! done in https://github.com/mattermost/mattermost/pull/30383/commits/eb5166a9f73a0a876de35bbe96c9c175a1bfe4ec